pkg/account/delete/client.go

```diff
@@ -50,23 +52,104 @@ func NewAccountAPIClient(accountUUID string, restClient *accounts.Client) Client
 	}
 }
 
-// NotFoundErr is a sentinel error signifying that the resource desired to be deleted was not found. Generally this error can be treated as a succeful "deletion" of the resource.
-var NotFoundErr = errors.New("nothing with given name found")
+// NotFoundError is an error signifying that the desired resource was not found. Generally this error can be treated as a successful "deletion" of the resource.
+type ResourceNotFoundError struct {
+	Identifier string
+}
+
+func (e *ResourceNotFoundError) Error() string {
+	return fmt.Sprintf("resource '%s' not found", e.Identifier)
+}
 
 // DeleteUser removes the user with the given email from the account
 // Returns error if any API call fails unless the user is already not present (HTTP 404)
 func (c *AccountAPIClient) DeleteUser(ctx context.Context, email string) error {
 	resp, err := c.client.UserManagementAPI.RemoveUserFromAccount(ctx, c.accountUUID, email).Execute()
 	defer closeResponseBody(resp)
 	if resp != nil && resp.StatusCode == 404 {
-		return NotFoundErr
+		return &ResourceNotFoundError{Identifier: email}
 	}
 	if err := handleClientResponseError(resp, err, fmt.Sprintf("failed to delete user %q", email)); err != nil {
 		return err
 	}
 	return nil
 }
 
+// DeleteServiceUser retrieves all service users, looks up the service user by name and removes it from the account.
+// Returns error if any API call fails unless the user is already not present, either in the list when looking by name or during the subsequent delete call.
+// In addition, an error is returned if multiple service users are found with the same name.
+func (c *AccountAPIClient) DeleteServiceUser(ctx context.Context, name string) error {
+	uid, err := c.getServiceUserIDByName(ctx, c.accountUUID, name)
+	if err != nil {
+		return err
+	}
+
+	resp, err := c.client.ServiceUserManagementAPI.DeleteUser(ctx, c.accountUUID, uid).Execute()
+	defer closeResponseBody(resp)
+	if resp != nil && resp.StatusCode == 404 {
+		return &ResourceNotFoundError{Identifier: name}
+	}
+	if err := handleClientResponseError(resp, err, fmt.Sprintf("failed to delete service user %q", name)); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (c *AccountAPIClient) getServiceUserIDByName(ctx context.Context, accountUUID, name string) (string, error) {
+	serviceUsers, err := c.getServiceUsers(ctx, accountUUID)
+	if err != nil {
+		return "", err
+	}
+
+	uid := ""
+	for _, s := range serviceUsers {
+		if s.Name == name {
+			if uid != "" {
+				return "", fmt.Errorf("found multiple service users with name '%s'", name)
+			}
+			uid = s.Uid
```

Comment:
Could you return already here? Simplifies the method return at the bottom.

Reply:
An earlier return is not possible because we must check for multiple matches (which should produce an error a couple of lines above)